### PR TITLE
feat: /ref/hash/shape links

### PR DESCRIPTION
**Pull request by tpluscode**

### CODE REVIEW

File: trifid/redirect.js [hunk 1]
```diff
@@ -53,7 +53,10 @@ function factory () {
         } else {
           versionPath = tags[0].name
         }
+      } else if (ref) {
+        versionPath = ref
       }
```
Comment (ludovicm67) [inline trifid/redirect.js:58]: Let's handle the case where a version is defined:

```suggestion
      } else if (version) {
        versionPath = version
      }
```

**Review by ludovicm67 — CHANGES_REQUESTED** [review]
There were a few issues that were breaking the links to shapes:
- refs were not caught, as it was introducing an extra path layer
- specific versions were not handled ; it was returning the last one instead

With the suggestions I made, it should work.

To try it locally, just install the dependencies using:

```sh
npm install
```

Run the Trifid instance by using:

```sh
npm run trifid
```

And you can run the following cUrl commands:

```sh
# latest
curl http://localhost:8080/latest/shape/standalone-cube-constraint

# get a specific version (you can see the old base)
curl http://localhost:8080/v0.0.1/shape/standalone-cube-constraint

# get current state from main branch
curl http://localhost:8080/ref/main/shape/standalone-cube-constraint
```

File: trifid/redirect.js [hunk 2]
```diff
@@ -32,11 +32,11 @@ function factory () {
       return res.redirect('/relation/')
     }
 
-    // Redirect to a version of a shape (for routes: `/vX.Y.Z/shape/NAME` and `/latest/shape/NAME`)
+    // Redirect to a version of a shape (for routes: `/vX.Y.Z/shape/NAME`, `/ref/COMMIT_ID/shape/NAME` and `/latest/shape/NAME`)
     const shapePath = requestPath.split('/').slice(3).join('/')
```
[the remark this answers is not in the document]
Comment (tpluscode) [reply trifid/redirect.js]: > But this means also that we will lose the feature of serving a shape that is from a subdirectory

I addressed that by adding the path a another capture in the regular expression

[the remark this answers is not in the document]
Comment (ludovicm67) [reply trifid/redirect.js]: Looks good! 👍

**Review by ludovicm67 — APPROVED** [review]

**Pull request merged**